src/com/esotericsoftware/kryo/GenericsResolver.java

```diff
@@ -0,0 +1,34 @@
+package com.esotericsoftware.kryo;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+
+/** Helper class that resolves a type name variable to a concrete class using the current class serialization stack
+ *
+ * @author Jeroen van Erp <jeroen@hierynomus.com> */
+public class GenericsResolver {
```

Comment:
This class is currently declared public. Is GenericResolver going to be exposed to the users? 
If yes, why and how? What would be the API?
If not, it should not be public I guess.